### ISSUE TITLE
[GraphQL] does not respect serialized_name option?

### ISSUE THREAD

**Issue by xdrew** [api-platform/api-platform#1205]
I have Restaurant resource that has some renamed fields in serialization yaml:
```
App\Entity\Restaurant:
  attributes:
    ...
    pictureUrl:
      serialized_name: picture
      groups: ['restaurant:read', 'restaurant:write']
    openPeriods:
      serialized_name: workHours
      groups: ['restaurant:read', 'restaurant:write']
      max_depth: 1
```

Rest queries are entirely ok with that, but graphql schema builder does not generate types for those properties. And I cannot use workHours, picture etc in my graphql queries.
I tried adding following config to resource.yaml:
```
resources;
  App\Entity\Restaurant:
    properties:
      workHours:
        subresource: {collection: true, resourceClass: App\Entity\OpenPeriod, maxDepth: 1}
        description: 'The dummy foo'
        readable: true
        writable: true
        readableLink: false
        writableLink: false
        required: false
```
Then propertyNameCollectionFactory was able to recognize new property, but propertyMetadataFactory returned empty type in getResourceObjectTypeFields method.

So, the questions are: 
1. Is this a correct behavior?
2. What is the preferred way of dealing with such properties?

**Comment by alanpoulain** [comment]
Yes, the GraphQL schema is not built by using the serialization configuration but by using the resource metadata.
Maybe you could do what you want by decorating the `PropertyNameCollectionFactory` which is used to determine the properties of a resource.

**Comment by alanpoulain** [comment]
Fixed in 2.6.
